packages/web-config-server/src/apiV1/dataBuilders/generic/table/tableOfDataValues/helpers/addSummaryToTable.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Tupaia Config Server
+ * Copyright (c) 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { checkValueSatisfiesCondition } from '@tupaia/utils';
+
+const percentageOfConditions = (valueSet, config) => {
+  const valueAllUndefined = valueSet.every(value => value === undefined);
+  if (valueSet.length === 0 || valueAllUndefined) return false;
```

Comment:
(Detail) This check is slightly quicker: just one non `undefined` value is enough to do calculations

```suggestion
  const isAnyValueDefined = valueSet.some(value => value !== undefined);
  if (valueSet.length === 0 || !isAnyValueDefined) return false;
```